### PR TITLE
Feat/password reset linking and updating of user data

### PR DESCRIPTION
**Pull request by aeschi**

### CODE REVIEW

File: src/auth/auth-store.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { create } from "zustand";
-import { Session } from "@supabase/supabase-js";
+import { Session, User } from "@supabase/supabase-js";
 import { supabaseClient } from "./supabase-client.ts";
```
Comment (ff6347) [inline src/auth/auth-store.tsx:3]: Extensions or not? currently we don't use them

File: src/auth/auth-store.tsx [hunk 2]
```diff
@@ -42,6 +45,14 @@ export const useAuthStore = create<AuthState>()((set, get) => {
       return !!get().session;
```
Comment (ff6347) [inline src/auth/auth-store.tsx:45]: Double negation is to much for my brain. Can you make this more readable

File: src/auth/auth-store.tsx [hunk 3]
```diff
@@ -134,10 +145,61 @@ export const useAuthStore = create<AuthState>()((set, get) => {
...
+        if (
+          window.confirm(
+            'Dein Passwort wurde geändert. Klicke auf "ok" um zu deinem Profil zu kommen.',
+          )
+        ) {
+          window.location.href = "/profile";
+        }
       }
 
       console.log("Update password success:", data);
```
Comment (ff6347) [inline src/auth/auth-store.tsx:157]: console.log should not land in production

File: src/auth/auth-store.tsx [hunk 4]
```diff
@@ -134,10 +145,61 @@ export const useAuthStore = create<AuthState>()((set, get) => {
...
     },
+
+    updateEmail: async (email: string) => {
+      const { data, error } = await supabaseClient.auth.updateUser({
+        email: email,
+      });
+
+      console.log("email:", data);
```
Comment (ff6347) [inline src/auth/auth-store.tsx:165]: same same

File: src/components/input/input-field-icon.tsx [hunk 5]
```diff
@@ -1,65 +1,57 @@
-import React, { Dispatch, SetStateAction } from "react";
+import React from "react";
 import { useI18nStore } from "../../i18n/i18n-store.ts";
 
 export interface InputFieldIconProps {
   label: string | React.ReactNode;
-  placeholder: string;
+  placeholder: string | undefined;
```
Comment (ff6347) [inline src/components/input/input-field-icon.tsx:6]: `placeholder?: string;`

File: src/components/input/input-field-icon.tsx [hunk 6]
```diff
@@ -1,65 +1,57 @@
...
+          <p className="italic">{placeholder}</p>
+          <button
+            className="self-end text-gdk-blue enabled:hover:text-gdk-light-blue"
+            onClick={onClick}
+          >
+            {icon}
+          </button>
+        </div>
+      )}
+      {!disabled && (
```
Comment (ff6347) [inline src/components/input/input-field-icon.tsx:37]: If not disabled? (just a style question but parsing the logic makes this harder IMO)

File: src/components/profile/profile-logged-in/profile-details.tsx [hunk 7]
```diff
@@ -1,47 +1,36 @@
...
         label={i18n.navbar.profile.settings.username}
-        placeholder={i18n.navbar.profile.settings.placeholderUser}
-        onClick={() => {
-          setUsernameDisabled(!usernameIsDisabled);
-        }}
-        disabled={usernameIsDisabled}
-        setDisabled={setUsernameDisabled}
-        icon={<EditIcon />}
-      ></InputFieldIcon>
+        placeholder={getUserData()?.user_metadata.signup_username}
```
Comment (ff6347) [inline src/components/profile/profile-logged-in/profile-details.tsx:21]: Your hooks throw errors but the errors are not handled here.

File: src/components/profile/profile-logged-in/profile-details.tsx [hunk 8]
```diff
@@ -1,47 +1,36 @@
...
         label={i18n.navbar.profile.settings.email}
-        placeholder={i18n.navbar.profile.settings.placeholderMail}
-        onClick={() => {
-          setEmailDisabled(!emailIsDisabled);
-        }}
-        disabled={emailIsDisabled}
-        setDisabled={setEmailDisabled}
-        icon={<EditIcon />}
-      ></InputFieldIcon>
+        placeholder={getUserData()?.email}
```
Comment (ff6347) [inline src/components/profile/profile-logged-in/profile-details.tsx:26]: unhandled possible errors

File: src/components/profile/profile-logged-in/profile-details.tsx [hunk 9]
```diff
@@ -50,10 +39,12 @@ const ProfileDetails: React.FC = () => {
             value="1234567890"
           />
           <TertiaryButton
-            onClick={() => {}}
+            onClick={() => {
+              forgotPassword(getUserData()?.email ?? "");
```
Comment (ff6347) [inline src/components/profile/profile-logged-in/profile-details.tsx:43]: unhandled possible error

File: src/components/profile/validation/email-input-with-icon.tsx [hunk 10]
```diff
@@ -0,0 +1,67 @@
+import React, { useState } from "react";
+import { useI18nStore } from "../../../i18n/i18n-store.ts";
```
Comment (ff6347) [inline src/components/profile/validation/email-input-with-icon.tsx:2]: Extensions…

File: src/components/profile/validation/email-input-with-icon.tsx [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+import EditIcon from "../../icons/edit-icon.tsx";
+import { useAuthStore } from "../../../auth/auth-store.tsx";
+
+export interface EmailInputWithIconProps {
+  label: string | React.ReactNode;
+  placeholder: string | undefined;
```
Comment (ff6347) [inline src/components/profile/validation/email-input-with-icon.tsx:8]: placeholder?: string

File: src/auth/auth-store.tsx [hunk 12]
```diff
@@ -14,11 +14,14 @@ interface RegistrationCredentials extends Credentials {
 interface AuthState {
   session: Session | null | undefined;
   isLoggedIn: () => boolean | undefined;
+  getUserData: () => User | undefined;
   login: ({ email, password }: Credentials) => void;
```
Comment (ff6347) [inline src/auth/auth-store.tsx:18]: login, register, updatePassword and some more are async and should return `Promise<void>`

File: src/components/profile/validation/email-input-with-icon.tsx [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+            className="self-end text-gdk-blue enabled:hover:text-gdk-light-blue"
+            onClick={() => {
+              setEmailDisabled(!emailIsDisabled);
+            }}
+          >
+            <EditIcon />
+          </button>
+        </div>
+      )}
+      {!emailIsDisabled && (
```
Comment (ff6347) [inline src/components/profile/validation/email-input-with-icon.tsx:36]: Double negation is not disabled…

File: src/components/profile/validation/username-input-with-icon.tsx [hunk 14]
```diff
@@ -0,0 +1,136 @@
+import React, { useCallback, useState } from "react";
+import { useI18nStore } from "../../../i18n/i18n-store.ts";
```
Comment (ff6347) [inline src/components/profile/validation/username-input-with-icon.tsx:2]: Extensions

File: src/components/profile/validation/username-input-with-icon.tsx [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+import EditIcon from "../../icons/edit-icon";
+import { UsernameErrors } from "./types";
+import Warning from "./warning";
+import { checkIfUsernameIsTaken, validateUsername } from "./validation";
+import { useAuthStore } from "../../../auth/auth-store.tsx";
+
+export interface UsernameInputWithIconProps {
+  label: string | React.ReactNode;
+  placeholder: string | undefined;
```
Comment (ff6347) [inline src/components/profile/validation/username-input-with-icon.tsx:11]: placeholder?: string

File: src/components/profile/validation/username-input-with-icon.tsx [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+      setUsernameErrors(usernameErrors);
+
+      const hasErrors = Object.values(usernameErrors).some((error) => !error);
+
+      if (hasErrors) {
+        target.setCustomValidity("Bitte überprüfe Deine Eingabe");
+        return;
+      }
+
+      const isTaken = await checkIfUsernameIsTaken(target.value);
```
Comment (ff6347) [inline src/components/profile/validation/username-input-with-icon.tsx:59]: Can this throw?

**Review by ff6347 — CHANGES_REQUESTED** [review]

File: src/components/profile/profile-logged-in/edit-password.tsx [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+          type="password"
+          value="1234567890"
+        />
+        <TertiaryButton
+          onClick={() => {
+            forgotPassword(getUserData()?.email ?? "");
+          }}
+          label={i18n.navbar.profile.settings.changePassword}
+        ></TertiaryButton>
+      </form>
```
Comment (raphael-arce) [inline src/components/profile/profile-logged-in/edit-password.tsx:29]: we don't need a form here, let's remove it 👍

File: src/components/profile/profile-logged-in/profile-details.tsx
Comment (raphael-arce) [inline src/components/profile/profile-logged-in/profile-details.tsx:1]: as discussed let's put this in a folder with its 3 subcomponents 👍

**Review by raphael-arce — CHANGES_REQUESTED** [review]
a couple suggestions 👍

**Pull request merged**